certbot/tests/log_test.py

```diff
@@ -27,12 +26,33 @@ def _call(cls, *args, **kwargs):
         return pre_arg_parse_setup(*args, **kwargs)
 
     def test_it(self):
-        with mock.patch('certbot.log.except_hook') as mock_except_hook:
-            with mock.patch('certbot.log.sys') as mock_sys:
-                self._call()
-
+        with mock.patch('certbot.log.util.atexit_register') as mock_register:
```

Comment:
If you're REALLY curious you can read [PEP 343](https://www.python.org/dev/peps/pep-0343/) which describes the motivation behind the `with` statement and how it works, but it's really more like a `try ... finally` block. For the file example, it's equivalent to something like:
```
f = open('foo.txt', 'w')
try:
    # imagine I'm the body of the with statement
finally:
    f.close()
```
Note this isn't exactly correct and there's code that describes how to exactly how to mimic the behavior in the above PEP.